### PR TITLE
STM32: Fix SD EEPROM emulation conflict

### PR DESCRIPTION
**Pull request by tpruvot**
current use of the card object may affect prints from sdcard

### CODE REVIEW

**Comment by marcio-ao** [comment]
@thinkyhead: Just to expand on this a bit, @tpruvot and I were discussing a problem where 'card.openFile' replaces the LCD message with the filename and hides the "Printer ready" message when called at startup to read EEPROM data on the STM32 chips. This PR resolves that issue, but there also is another reason for it:

The previous code was reading a binary EEPROM file using `card.openFile`. That function is for reading GCODE files and changes Marlin's state regarding the file that is currently printing. If an M500, M501, M502 were issued during a print, causing EEPROM access, it likely would kill a print. This PR gets around the problem by using a new File object independent of what Marlin is using for printing.

**Comment by thinkyhead** [comment]
>  @tpruvot and I were discussing a problem where 'card.openFile' replaces the LCD message with the filename and hides the "Printer ready" message when called at startup to read EEPROM data on the STM32 chips.

That's the right way to do it. The `CardReader` methods are the high level interface to the file system. The class maintains one `SDFile` object for the file being printed, another for the power-loss recovery file, and so on.

Other classes can and should bypass the `CardReader` class and use their own `SDFile` objects when needed. Methods for Power Loss Recovery currently in `cardreader.cpp` don't really need to be there, and those will be moved to the `PrintJobRecovery` class. I recommend moving all `SDFile` code being created for this PR to the `MarlinSettings` class or to a HAL supporting class.

**Comment by thinkyhead** [comment]
On reëxamining the code I see that the pieces _are_ being added to the supporting class, so it's good!

File: Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp [hunk 1]
```diff
@@ -64,7 +64,7 @@ static char HAL_STM32F1_eeprom_content[HAL_STM32F1_EEPROM_SIZE];
     if (!card.isDetected()) return false;
 
     SdFile file, root = card.getroot();
-    if (!file.open(&root, EEPROM_FILENAME, O_CREAT | O_WRITE | O_TRUNC))
+    if (file.open(&root, EEPROM_FILENAME, O_CREAT | O_WRITE | O_TRUNC))
```
Comment (tpruvot) [inline Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp:67]: hmm why did you change that ?

Comment (marcio-ao) [reply Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp:67]: @thinkyhead: `file.open` returns true on success, so in fact the code was correct before you changed it.

Comment (thinkyhead) [reply Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp:67]: Typo. I was going to do something else there and apparently didn't finish.

Comment (tpruvot) [reply Marlin/src/HAL/HAL_STM32F1/persistent_store_sdcard.cpp:67]: yes, but you merged it with that, could you fix it on the git ?
